evap/rewards/views.py

```diff
@@ -22,28 +22,54 @@
     RewardPointRedemptionEvent,
     SemesterActivation,
 )
-from evap.rewards.tools import grant_eligible_reward_points_for_semester, reward_points_of_user, save_redemptions
+from evap.rewards.tools import (
+    grant_eligible_reward_points_for_semester,
+    redeemed_points_of_user,
+    reward_points_of_user,
+    save_redemptions,
+)
 from evap.staff.views import semester_view
 
 
+def check_consistent_previous_redemption_counts(request):
+    reward = request.POST.get("previous_reward_points")
+    redeemed = request.POST.get("previous_redeemed_points")
+    if reward is None or redeemed is None or not reward.isalnum() or not redeemed.isalnum():
+        raise BadRequest("Invalid redeemed-points or left-points field in redemption request")
+    return int(reward) == reward_points_of_user(request.user) and int(redeemed) == redeemed_points_of_user(request.user)
+
+
+def redeem_reward_points(request):
+    redemptions = {}
+    try:
+        for key, value in request.POST.items():
+            if key.startswith("points-"):
+                event_id = int(key.rpartition("-")[2])
+                redemptions[event_id] = int(value)
+    except ValueError as e:
+        raise BadRequest from e
+
+    try:
+        save_redemptions(request, redemptions)
+        messages.success(request, _("You successfully redeemed your points."))
+    except (NoPointsSelected, NotEnoughPoints, RedemptionEventExpired) as error:
+        messages.warning(request, error)
+
+
 @reward_user_required
 def index(request):
+    status = 200
     if request.method == "POST":
-        redemptions = {}
-        try:
-            for key, value in request.POST.items():
-                if key.startswith("points-"):
-                    event_id = int(key.rpartition("-")[2])
-                    redemptions[event_id] = int(value)
-        except ValueError as e:
-            raise BadRequest from e
-
-        try:
-            save_redemptions(request, redemptions)
-            messages.success(request, _("You successfully redeemed your points."))
-        except (NoPointsSelected, NotEnoughPoints, RedemptionEventExpired) as error:
-            messages.warning(request, error)
-
+        if check_consistent_previous_redemption_counts(request):
+            redeem_reward_points(request)
+        else:
+            messages.warning(
+                request,
+                _(
+                    "Probably your browser sent multiple redemption request. You can see all successful redemptions below."
```

Comment:
```suggestion
                    "It appears that your browser sent multiple redemption requests. You can see all successful redemptions below."
```

Reply:
Also, I turned the warning into an error, as the requested redemption failed